### PR TITLE
feat: Implement SfColorPicker in SfProductCard

### LINKED ISSUE #881: [FEATURE] Implement SfColorPicker in SfProductCard
**Describe how this feature will help us improve**
This issue is blocked by https://github.com/DivanteLtd/storefront-ui/issues/851
With new design, we need to implement SfColorPicker in SfProductCard

**Additional context**
Mobile: https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-Design-System?node-id=264%3A566
Desktop: https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-Design-System?node-id=771%3A306

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue
Closes #881 

# Scope of work
SfColor Picker added to SfProductCard

# Screenshots of visual changes
Zrzut ekranu z 2020-10-26 14-37-51
Zrzut ekranu z 2020-10-26 14-39-05
Zrzut ekranu z 2020-10-26 14-43-09

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 1]
```diff
@@ -83,13 +83,44 @@ storiesOf("Organisms|ProductCard", module)
...
       },
+      handleSelectedColor(color) {
+        this.colors.map((el) => {
+          if (el.label === color.label) {
+            el.selected = !el.selected;
+          } else {
+            el.selected = false;
+          }
+        });
+      },
```
Comment (AdamPawlinski) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:94]: I wonder if it's necessary here? You have already handled selecting color inside SfProductCard component. Here I would add sth like this: 
``` 
alertColor(event) {alert(event.value);}
```
and then inside template:
```
 @click:colors="alertColor($event)"
```
To show that emitting event works properly like with other events. What do you think?
By the way there is a bug in storybook in emitting add to cart event, but I will handle that :)

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 2]
```diff
@@ -181,13 +213,42 @@ storiesOf("Organisms|ProductCard", module)
       alert(label) {
         alert(label);
       },
+      handleSelectedColor(color) {
+        this.colors.map((el) => {
+          if (el.label === color.label) {
+            color.selected = !color.selected;
+          }
+        });
+      },
```
Comment (AdamPawlinski) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:222]: The same like above

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 3]
```diff
@@ -279,13 +341,42 @@ storiesOf("Organisms|ProductCard", module)
...
```
Comment (AdamPawlinski) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:350]: The same like above

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 4]
```diff
@@ -385,19 +477,44 @@ storiesOf("Organisms|ProductCard", module)
...
```
Comment (AdamPawlinski) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:509]: The same like above

**Review by AdamPawlinski — CHANGES_REQUESTED** [review]
Just one thing in story :) Except that it is great work as always :)

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js (hunk 1 above)
Comment (justyna-13) [reply packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:94]: I think it's nice to have it. Thanks to it you can see how it really works - you can change color - and see a way how to implement it so one step easier :)

**Comment by Baroshem** [comment]
@justyna-13 what is the relation between this PR and https://github.com/vuestorefront/storefront-ui/pull/1518? They both seem to do the same thing.

File: packages/shared/styles/components/organisms/SfProductCard.scss [hunk 5]
```diff
@@ -37,6 +37,23 @@
     text-decoration: none;
     margin: var(--product-card-link-margin, 0);
   }
+  &__colors {
+    --color-picker-label-font-size: var(--font-size--base);
```
Comment (Baroshem) [inline packages/shared/styles/components/organisms/SfProductCard.scss:41]: Would it be possible to somehow get rid of these CSS variables?

Comment (justyna-13) [reply packages/shared/styles/components/organisms/SfProductCard.scss:41]: this will be done in separate PR

**Review by Baroshem — APPROVED** [review]
Please, resolve conflicts and you are ready to go ;)

**Review by Baroshem — APPROVED** [review]
Like it!

**Comment by Baroshem** [comment]
@justyna-13 

What is left to do here is to resolve conflicts and you are good to merge it :)

**Comment by justyna-13** [comment]
conflicts resolved but can't merge it - there are requested changes that block it ;) @AdamPawlinski can you check it and see if we can merge it?

**Review by AdamPawlinski — APPROVED** [review]

**Comment by AdamPawlinski** [comment]
@justyna-13 Approved, you can merge it :) thanks for great work.

**Pull request merged**